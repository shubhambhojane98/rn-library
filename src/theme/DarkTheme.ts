import type {Theme} from '../utils/types';

let DarkTheme: Theme;

DarkTheme = {
  dark: true,
  colors: {
    primary: '#BB86FC',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#262833',
    error: '#CF6679',
    textColor: '#FFF',
    shadowColor: '#D7D7D7',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
  },
};

export default DarkTheme;
