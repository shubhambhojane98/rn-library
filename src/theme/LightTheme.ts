import type {Theme} from '../utils/types';

let LightTheme: Theme;

LightTheme = {
  dark: false,
  colors: {
    primary: '#6200EE',
    primaryVariant: '#3700B3',
    secondary: '#03DAC6',
    secondaryVariant: '#018786',
    background: '#FFF',
    error: '#B00020',
    textColor: '#000',
    shadowColor: '#222',
    black: '#000',
    white: '#FFF',
    lightgrey: '#D7D7D7',
    backgroundLayout: 'rgba(0, 0, 0, 0.8)',
    transparent: 'transparent',
    linkColor: '#0000EE',
    greyOpac50: 'rgba(53, 53, 53, 0.5)',
    greyOpac10: 'rgba(53, 53, 53, 0.1)',
  },
};

export default LightTheme;
