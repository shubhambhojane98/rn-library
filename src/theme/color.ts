import {Appearance} from 'react-native';

interface colors {
  yellow: string;
  black: string;
  red: string;
  violet: string;
  taupe: string;
  beige: string;
  grey: string;
  lightgrey: string;
  blue: string;
  darkGrey: string;
  white: string;
  transportation: string;
  logistic: string;
  environment: string;
  sports: string;
  healthcare: string;
  background: string;
  textColor: string;
  placeholderTextColor: string;
  switchBackgroundColor: string;
  paginationDotColor: string;
  paginationDotBgColor: string;
  blackOpac80: string;
  blackOpac60: string;
  blackOpac40: string;
  blackOpac20: string;
  greyOpac80: string;
  greyOpac60: string;
  greyOpac50: string;
  greyOpac40: string;
  greyOpac20: string;
  greyOpac10: string;
  switchActiveBgColor: string;
  switchInActiveBgColor: string;
  switchKnobColor: string;
}

const primaryColors = {
  yellow: '#FAD000',
  black: '#231F20',
  red: '#A81E23',
  violet: '#533467',
  taupe: '#695958',
  beige: '#C3B59F',
  grey: '#353535',
  white: '#fff',
  lightgrey: '#D7D7D7',
  blue: '#007BC1',
  darkGrey: 'rgba(0, 0, 0, 0.2)',
};

const opacColors = {
  blackOpac80: 'rgba(0, 0, 0, 0.8)',
  blackOpac60: 'rgba(0, 0, 0, 0.6)',
  blackOpac40: 'rgba(0, 0, 0, 0.4)',
  blackOpac20: 'rgba(0, 0, 0, 0.2)',
  greyOpac80: 'rgba(53, 53, 53, 0.8)',
  greyOpac60: 'rgba(53, 53, 53, 0.6)',
  greyOpac50: 'rgba(53, 53, 53, 0.5)',
  greyOpac40: 'rgba(53, 53, 53, 0.4)',
  greyOpac20: 'rgba(53, 53, 53, 0.2)',
  greyOpac10: 'rgba(53, 53, 53, 0.1)',
};

const categoryColors = {
  transportation: '#007BC1',
  logistic: '#EF4065',
  environment: '#009A46',
  sports: '#F26524',
  healthcare: '#00ABBE',
};

let themeColors: colors;

const colorScheme = Appearance.getColorScheme();
if (colorScheme === 'dark') {
  // Use dark color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.black,
    textColor: primaryColors.white,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.white,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
} else {
  // Use light color scheme
  themeColors = {
    ...primaryColors,
    ...categoryColors,
    ...opacColors,
    background: primaryColors.yellow,
    textColor: primaryColors.black,
    placeholderTextColor: '#5D5D5D',
    switchBackgroundColor: primaryColors.lightgrey,
    paginationDotColor: primaryColors.white,
    paginationDotBgColor: primaryColors.darkGrey,
    switchActiveBgColor: primaryColors.yellow,
    switchInActiveBgColor: primaryColors.white,
    switchKnobColor: primaryColors.black,
  };
}

export default themeColors;
