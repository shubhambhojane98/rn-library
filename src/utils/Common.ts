export const defaultScale = 0.2;
