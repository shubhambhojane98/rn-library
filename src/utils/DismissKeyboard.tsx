import React from 'react';
import {TouchableWithoutFeedback, Keyboard} from 'react-native';
//=========for dismissing the keyboard when taps on the screen =========
interface DismissKeyBoardProps {
  children: any;
}
const DismissKeyBoard: React.FC<DismissKeyBoardProps> = ({children}) => {
  return (
    <TouchableWithoutFeedback onPress={() => Keyboard.dismiss()}>
      {children}
    </TouchableWithoutFeedback>
  );
};
export {DismissKeyBoard};
