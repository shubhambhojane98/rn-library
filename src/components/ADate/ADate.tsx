import React, {useState} from 'react';
import {View, TouchableOpacity, StyleSheet} from 'react-native';
import {moderateScale} from 'react-native-size-matters';
import DateTimePicker from '@react-native-community/datetimepicker';
import ATextInput from '../../components/ATextInput/ATextInput';
import {defaultScale} from '../../utils/Common';

interface Prop {
  onSelectedDate: (fDate: any) => void;
  height?: number;
  borderColor?: string;
  placeholder?: string;
  disable?: boolean;
  errorText?: string;
  marginBottom?: number;
  marginLeft?: number;
  marginRight?: number;
  marginTop?: number;
}

const TDate: React.FC<Prop> = ({
  borderColor,
  placeholder,
  disable,
  errorText,
  marginBottom = 0,
  marginLeft = 0,
  marginRight = 0,
  marginTop = 0,
  onSelectedDate,
}) => {
  const [show, setShow] = useState(false);
  const getFormatedDate = () => {
    let todayDate = new Date();
    let fDate =
      todayDate.getDate() +
      '/' +
      (todayDate.getMonth() + 1) +
      '/' +
      todayDate.getFullYear();
    return fDate;
  };
  const [date, setDate] = useState(getFormatedDate());

  const OpenCalender = () => {
    if (!disable) {
      setShow(true);
    }
  };
  const onChange = (event: any, selectedDate: any) => {
    setShow(false);

    const currentDate = selectedDate;
    let tempDate = new Date(currentDate);
    let fDate =
      tempDate.getDate() +
      '/' +
      (tempDate.getMonth() + 1) +
      '/' +
      tempDate.getFullYear();
    setDate(fDate);
    onSelectedDate(fDate);
  };
  return (
    <View renderToHardwareTextureAndroid style={styles.containerStyle}>
      <View>
        <ATextInput
          placeholder={placeholder}
          editable={false}
          value={date}
          disable={disable}
          errorText={errorText}
          rightIcon={'Calender'}
          borderColor={borderColor}
          marginBottom={marginBottom}
          marginLeft={marginLeft}
          marginRight={marginRight}
          marginTop={marginTop}
        />
        <TouchableOpacity
          activeOpacity={0.9}
          onPress={OpenCalender}
          style={styles.dropDownButton}
        />
        {show && (
          <DateTimePicker
            testID="DateTimePicker"
            value={new Date()}
            display="default"
            onChange={onChange}
          />
        )}
      </View>
    </View>
  );
};

export default TDate;

const styles = StyleSheet.create({
  containerStyle: {
    marginRight: moderateScale(20, defaultScale),
    marginLeft: moderateScale(20, defaultScale),
    marginBottom: moderateScale(20, defaultScale),
    marginTop: moderateScale(20, defaultScale),
  },
  dropDownButton: {
    position: 'absolute',
    top: 0,
    left: 0,
    width: '100%',
    height: '100%',
  },
});
