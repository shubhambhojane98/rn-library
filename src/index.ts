import {defaultScale} from './utils/Common';
import {consoleLog, consoleWarn, consoleInfo} from './utils/Log';
import {CheckboxStatus} from './components/ACheckbox/ACheckboxEnum';
import {Color} from './theme';
export {default as AButton} from './components/AButton/AButton';
export {default as ATypography} from './components/ATypography/ATypography';
export {default as AAvatar} from './components/AAvatar/AAvatar';
export {default as ABadge} from './components/ABadge/ABadge';
export {default as AImage} from './components/AImage/AImage';
export {default as ARadioButton} from './components/ARadioButton/ARadioButton';
export {default as ARadioButtonGroup} from './components/ARadioButtonGroup/ARadioButtonGroup';
export {default as ATextInput} from './components/ATextInput/ATextInput';
export {default as ASwitch} from './components/ASwitch/ASwitch';
export {default as AVirtualizedList} from './components/AVirtualizedList/AVirtualizedList';
export {default as AModal} from './components/AModal/AModal';
export {default as ASwitchSelector} from './components/ASwitchSelector/ASwitchSelector';
export {default as AStarRating} from './components/AStarRating/AStarRating';
export {default as ACheckbox} from './components/ACheckbox/ACheckbox';
export {default as AToast} from './components/AToast/AToast';
export {default as ACard} from './components/ACard/ACard';
export {consoleLog, consoleWarn, consoleInfo};
export {Color};
export {CheckboxStatus};
