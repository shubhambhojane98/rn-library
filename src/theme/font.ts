import { Platform, PlatformOSType } from 'react-native';
import { Fonts } from '../utils/types';


export default function configureFonts(config?: {
    [platform in PlatformOSType | 'default']?: Fonts;
  }): Fonts {
    const fonts = Platform.select({ ...config }) as Fonts;
    return fonts;
  }