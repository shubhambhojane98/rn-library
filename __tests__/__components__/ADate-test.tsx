import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import ADate from '../../src/components/ADate/ADate';
import {Color} from '../../src/theme';
import {moderateScale} from 'react-native-size-matters';
import {defaultScale} from '../../src/utils/Common';

it('renders correctly with default props', () => {
  const tree = renderer.create(<ADate onSelectedDate={jest.fn()} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with borderColor, marginBottom , marginLeft , marginRight ,marginTop , onSelectedDate,', () => {
  const tree = renderer
    .create(
      <ADate
        borderColor={Color.environment}
        onSelectedDate={jest.fn()}
        marginRight={moderateScale(20, defaultScale)}
        marginLeft={moderateScale(20, defaultScale)}
        marginBottom={moderateScale(20, defaultScale)}
        marginTop={moderateScale(20, defaultScale)}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with disable & error text props', () => {
  const tree = renderer
    .create(
      <ADate
        disable={false}
        errorText={'wrong Date'}
        onSelectedDate={jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with disable & placeholder props', () => {
  const tree = renderer
    .create(
      <ADate
        disable={true}
        placeholder={'placeholder'}
        onSelectedDate={jest.fn()}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
