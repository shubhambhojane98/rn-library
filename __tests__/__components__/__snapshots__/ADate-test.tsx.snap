// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with borderColor, marginBottom , marginLeft , marginRight ,marginTop , onSelectedDate, 1`] = `
<View
  renderToHardwareTextureAndroid={true}
  style={
    Object {
      "marginBottom": 24.57142857142857,
      "marginLeft": 24.57142857142857,
      "marginRight": 24.57142857142857,
      "marginTop": 24.57142857142857,
    }
  }
>
  <View>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 2,
                "borderWidth": 1,
                "flexDirection": "row",
              },
              false,
              Object {
                "borderColor": "#009A46",
              },
              Object {
                "marginBottom": 38.61224489795918,
                "marginLeft": 38.61224489795918,
                "marginRight": 38.61224489795918,
                "marginTop": 38.61224489795918,
                "width": undefined,
              },
            ]
          }
        >
          <TextInput
            editable={false}
            onBlur={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#D7D7D7"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "height": 58.971428571428575,
                "paddingLeft": 18.428571428571427,
                "textAlign": "left",
              }
            }
            value="18/7/2022"
          />
          <View
            style={
              Object {
                "paddingLeft": 6.142857142857142,
                "paddingRight": 18.428571428571427,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 38.61224489795918,
            "marginRight": 38.61224489795918,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly with default props 1`] = `
<View
  renderToHardwareTextureAndroid={true}
  style={
    Object {
      "marginBottom": 24.57142857142857,
      "marginLeft": 24.57142857142857,
      "marginRight": 24.57142857142857,
      "marginTop": 24.57142857142857,
    }
  }
>
  <View>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 2,
                "borderWidth": 1,
                "flexDirection": "row",
              },
              false,
              Object {
                "borderColor": undefined,
              },
              Object {
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "width": undefined,
              },
            ]
          }
        >
          <TextInput
            editable={false}
            onBlur={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#D7D7D7"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "height": 58.971428571428575,
                "paddingLeft": 18.428571428571427,
                "textAlign": "left",
              }
            }
            value="18/7/2022"
          />
          <View
            style={
              Object {
                "paddingLeft": 6.142857142857142,
                "paddingRight": 18.428571428571427,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly with disable & error text props 1`] = `
<View
  renderToHardwareTextureAndroid={true}
  style={
    Object {
      "marginBottom": 24.57142857142857,
      "marginLeft": 24.57142857142857,
      "marginRight": 24.57142857142857,
      "marginTop": 24.57142857142857,
    }
  }
>
  <View>
    <View>
      <View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderRadius": 2,
                "borderWidth": 1,
                "flexDirection": "row",
              },
              false,
              Object {
                "borderColor": "#B00020",
              },
              Object {
                "marginBottom": 0,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 0,
                "width": undefined,
              },
            ]
          }
        >
          <TextInput
            editable={false}
            onBlur={[Function]}
            onFocus={[Function]}
            placeholderTextColor="#D7D7D7"
            style={
              Object {
                "color": "#000",
                "flex": 1,
                "height": 58.971428571428575,
                "paddingLeft": 18.428571428571427,
                "textAlign": "left",
              }
            }
            value="18/7/2022"
          />
          <View
            style={
              Object {
                "paddingLeft": 6.142857142857142,
                "paddingRight": 18.428571428571427,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Mulish-Regular",
                "fontSize": 17.2,
                "fontWeight": "normal",
              },
              Object {
                "color": "#B00020",
              },
              Object {
                "textAlign": "left",
              },
              undefined,
            ]
          }
          theme={
            Object {
              "colors": Object {
                "background": "#FFF",
                "backgroundLayout": "rgba(0, 0, 0, 0.8)",
                "black": "#000",
                "error": "#B00020",
                "greyOpac10": "rgba(53, 53, 53, 0.1)",
                "greyOpac50": "rgba(53, 53, 53, 0.5)",
                "lightgrey": "#D7D7D7",
                "linkColor": "#0000EE",
                "primary": "#FAD000",
                "primaryVariant": "#3700B3",
                "secondary": "#03DAC6",
                "secondaryVariant": "#018786",
                "shadowColor": "#222",
                "textColor": "#000",
                "transparent": "transparent",
                "white": "#FFF",
              },
              "dark": false,
            }
          }
        >
          wrong Date
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;

exports[`renders correctly with disable & placeholder props 1`] = `
<View
  renderToHardwareTextureAndroid={true}
  style={
    Object {
      "marginBottom": 24.57142857142857,
      "marginLeft": 24.57142857142857,
      "marginRight": 24.57142857142857,
      "marginTop": 24.57142857142857,
    }
  }
>
  <View>
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#D7D7D7",
              "borderRadius": 2,
              "flexDirection": "row",
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "width": undefined,
            }
          }
        >
          <TextInput
            editable={false}
            placeholder="placeholder"
            placeholderTextColor="#D7D7D7"
            style={
              Object {
                "flex": 1,
                "height": 58.971428571428575,
                "paddingLeft": 18.428571428571427,
                "textAlign": "left",
              }
            }
          />
          <View
            style={
              Object {
                "paddingLeft": 6.142857142857142,
                "paddingRight": 18.428571428571427,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "marginLeft": 0,
            "marginRight": 0,
          }
        }
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
